From 5bf798dc8e8492127189cb5b7debdf5681f24919 Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Sat, 22 Oct 2022 14:25:09 +0800
Subject: [PATCH] sw64: always restore MATCH configuration after scheduling
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 5bf798dc8e8492127189cb5b7debdf5681f24919
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I645N4

--------------------------------

On SW64, MATCH configuration is a part of task context which should
be switched after scheduling. If not, current task may trap because
of the MATCH configuration of previous task, which is unexpected.

Signed-off-by: He Sheng <hesheng@wxiat.com>
Reviewed-by: Cui Wei <cuiwei@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/ptrace.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/sw_64/kernel/ptrace.c b/arch/sw_64/kernel/ptrace.c
index e98679d10fae..10afa6936357 100644
--- a/arch/sw_64/kernel/ptrace.c
+++ b/arch/sw_64/kernel/ptrace.c
@@ -435,9 +435,6 @@ void restore_da_match_after_sched(void)
 	unsigned long dc_ctl;
 	struct pcb_struct *pcb = &task_thread_info(current)->pcb;
 
-	if (!(pcb->da_match || pcb->da_mask || pcb->dv_match || pcb->dv_mask || pcb->dc_ctl))
-		return;
-	printk("Restroe MATCH status, pid: %d\n", current->pid);
 	rwcsr(WCSR, CSR_DA_MATCH, 0);
 	rwcsr(WCSR, CSR_DA_MASK, pcb->da_mask);
 	rwcsr(WCSR, CSR_DA_MATCH, pcb->da_match);
-- 
2.33.0

